Drivers/Src/LTC681x.c

```diff
@@ -965,8 +963,10 @@ int8_t LTC681x_rdstat(uint8_t reg, //Determines which Stat  register is read bac
       if (received_pec != data_pec)
       {
         pec_error = -1;                             //The pec_error variable is simply set negative if any PEC errors
-        ic[c_ic].stat.pec_match[reg-1]=1;
-
+        if (reg < 3) {
+          //keep compiler happy and stop index outta bound errors
```

Comment:
imo this would be cooler if it was like
```
ASSERT (reg < 3);
ic[c_ic].stat.pec_match[reg-1]=1;
```
to make it clearer that `reg >= 3` should never happen, but what you have here is fine